### PR TITLE
Deploy "N" EDPM nodes

### PR DESCRIPTION
**Pull request by fao89**
ref: https://github.com/openstack-k8s-operators/dataplane-operator/pull/189

`make edpm_deploy DATAPLANE_TOTAL_NODES=4 DATAPLANE_SINGLE_NODE=false`

### CODE REVIEW

**Review by fultonj — COMMENTED** [review]
I normally just loop on the make command so I'm not sure this patch is necessary.
```
 for I in $(seq 0 $NODES); do
  make edpm_compute EDPM_COMPUTE_SUFFIX=$I
done
```
Did someone request it or does the CI need it?

If we're going to have this loop do we also want to add it to `edpm_compute_repos`? If we don't we'd just loop it anway.

nit: should the variable name of the loop `for number` be capitalized `for NUMBER`?

Overall, I think we need to determine if we need the patch to add the loop when the user can just use their own loop.

**Comment by fao89** [comment]
@fultonj I agree, I'll drop the loop.
About `edpm_compute_repos` instead of looping, I believe we can use a service:
https://github.com/openstack-k8s-operators/dataplane-operator/pull/189
the idea is making easier to do scale tests (OSP-23847), so instead of calling `edpm_compute_repos` 200 times, it would be better t have it set in parallel through AEE

File: devsetup/Makefile [hunk 1]
```diff
@@ -97,11 +98,27 @@ edpm_compute_repos: ## Configure EDPM compute VM with repositories
 	$(eval $(call vars))
 	scripts/edpm-compute-repos.sh ${EDPM_COMPUTE_SUFFIX}
 
+.PHONY: edpm_nodes
```
Comment (slagle) [inline devsetup/Makefile:101]: I think this target could probably just take the place of edpm_compute/edpm_compute_cleanup. No reason to have multiple ways of doing it. If you just want one node, you'd set EDPM_TOTAL_NODES=1.

File: devsetup/Makefile [hunk 2]
```diff
@@ -97,11 +98,27 @@ edpm_compute_repos: ## Configure EDPM compute VM with repositories
 	$(eval $(call vars))
 	scripts/edpm-compute-repos.sh ${EDPM_COMPUTE_SUFFIX}
```
Comment (slagle) [inline devsetup/Makefile:99]: I'd then add the loop here as well so that you don't need to call this target multiple times.

**Review by slagle — CHANGES_REQUESTED** [review]

File: scripts/gen-edpm-kustomize.sh [hunk 3]
```diff
@@ -71,6 +71,11 @@ patches:
     - op: replace
       path: /spec/nodes/edpm-compute-0/node/ansibleSSHPrivateKeySecret
       value: ${EDPM_ANSIBLE_SECRET}
+    - op: replace
+      path: /spec/roles/edpm-compute/services
+      value:
+      - repo-setup
```
Comment (fao89) [inline scripts/gen-edpm-kustomize.sh:77]: requires: https://github.com/openstack-k8s-operators/dataplane-operator/pull/189

**Review by fultonj — COMMENTED** [review]
Looks good to me but I didn't test it in my environment yet.

**Comment by fao89** [comment]
recheck

**Comment by fao89** [comment]
recheck

**Comment by fao89** [comment]
recheck

**Pull request merged**